beets/plugins.py

```diff
@@ -105,6 +104,14 @@ class BeetsPlugin(metaclass=abc.ABCMeta):
     the abstract methods defined here.
     """
 
+    _raw_listeners: ClassVar[dict[EventType, list[Listener]]] = defaultdict(
```

Comment:
Since you are at it anyways, could we make the raw listeners to a set? :upside_down_face: 

E.g. instead of:
```
ClassVar[dict[EventType, list[Listener]]]
```
```
ClassVar[dict[EventType, set[Listener]]]
```

Reply:
I considered this when I looked to remove this condition

```py
        if func not in self._raw_listeners[event]:
```

But decided against it since ordering may have been an intentional design choice here.